.github/workflows/autogpt-builder-docker.yml

```diff
@@ -0,0 +1,41 @@
+name: AutoGPT Builder Docker
+
+on:
+  push:
+    branches: [ master ]
+    paths:
+      - 'rnd/autogpt_builder/**'
+
+defaults:
+  run:
+    shell: bash
+    working-directory: rnd/autogpt_builder
+
+env:
+  PROJECT_ID: agpt-dev
+  IMAGE_NAME: agpt-builder-dev
+  REGION: us-central1
+
+jobs:
+  build-and-push:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up Cloud SDK
+        uses: google-github-actions/setup-gcloud@v0.2.1
+        with:
+          project_id: ${{ env.PROJECT_ID }}
+          service_account_key: ${{ secrets.GCP_SA_KEY }}
+          export_default_credentials: true
+
+      - name: Configure Docker
+        run: gcloud auth configure-docker ${{ env.REGION }}-docker.pkg.dev
+
+      - name: Build Docker image
```

Comment:
Does GitHub actions maintain an image cache for us or will this rebuild the image each time from scratch?